commit 7485adcf0513846bf7ad215cd07f7fac565b5d0a
Author: Tim Foster <tim.foster@joyent.com>
Date:   2019-08-08T16:53:01+01:00 (9 weeks ago)
    
    MANTA-4481 mako-gc-feeder download-sdcnode should move off download.joyent.com

diff --git a/tools/download-sdcnode b/tools/download-sdcnode
index 7a515ef..3ec9c8b 100755
--- a/tools/download-sdcnode
+++ b/tools/download-sdcnode
@@ -1,6 +1,7 @@
 #!/bin/bash
 
-BASEURL='https://download.joyent.com/pub/build/sdcnode'
+BASEURL='https://us-east.manta.joyent.com'
+MANTA_PATH='/Joyent_Dev/public/releng/sdcnode'
 BRANCH='master'
 IMAGE=$1
 VARIANT=$2
@@ -11,14 +12,32 @@ if [[ -z $IMAGE || -z $VARIANT || -z $OUTDIR ]]; then
 	exit 1
 fi
 
-FULLURL="$BASEURL/$IMAGE/$BRANCH-latest/sdcnode/"
+HAS_JSON=$(command -v json)
+if [[ -z "$HAS_JSON" ]]; then
+	printf 'ERROR: "json" command not found in $PATH'
+	exit 1
+fi
+
+FULLURL="$BASEURL/$MANTA_PATH/$IMAGE/$BRANCH-latest"
 TARGET="sdcnode-$VARIANT-$IMAGE.tar.gz"
 
 #
-# Download the index page which lists the current set of available sdcnode
-# builds:
+# We assume $FULLURL points to a Manta object that contains a path to
+# the actual branch Manta directory.
+#
+if ! manta_path=$(curl -sSfL "$FULLURL") || [[ -z "$manta_path" ]]; then
+	printf "ERROR: could not determine manta_path url for $FULLURL\n" >&2
+	exit 1
+else
+	FULLURL="$BASEURL/$manta_path/sdcnode"
+fi
+
+#
+# Now download the Manta directory index which lists the current set of
+# available sdcnode builds. Note that this is assumed to be an
+# application/x-json-stream with one json record per line.
 #
-if ! list=$(curl -sSfL "$FULLURL") || [[ -z "$list" ]]; then
+if ! list=$(curl -sSfL "$FULLURL" | json -ag name) || [[ -z "$list" ]]; then
 	printf 'ERROR: could not download index page\n' >&2
 	exit 1
 fi
@@ -31,7 +50,7 @@ fi
 #
 if ! name=$(awk -v "v=$VARIANT" -v "b=$BRANCH" -v "i=$IMAGE" -F\" '
     BEGIN { pattern = "^sdcnode-"v"-"i"-"b"-.*.tgz$"; }
-    $1 == "<a href=" && $2 ~ pattern { print $2 }' <<< "$list") ||
+    $1 ~ pattern { print $1 }' <<< "$list") ||
     [[ -z "$name" ]]; then
 	printf 'ERROR: could not locate file name in index page\n' >&2
 	exit 1
@@ -43,7 +62,7 @@ if [[ ! -f $OUTDIR/$name ]]; then
 	# download it now to a temporary file.  If it succeeds, move it into
 	# place.
 	#
-	if ! curl -sSf -o "$OUTDIR/.tmp.$name" "$FULLURL$name"; then
+	if ! curl -sSf -o "$OUTDIR/.tmp.$name" "$FULLURL/$name"; then
 		printf 'ERROR: could not download sdcnode' >&2
 		rm -f "$OUTPUT.tmp"
 		exit 1
